app/models/miq_retire_task/state_machine.rb

```diff
@@ -0,0 +1,35 @@
+module MiqRetireTask::StateMachine
+  extend ActiveSupport::Concern
+
+  def run_retire
+    signal :start_retirement
+  end
+
+  def start_retirement
+    if source.retired?
+      fail!("#{self.class.model_being_retired} already retired")
+    elsif source.retiring?
+      fail!("#{self.class.model_being_retired} already in the process of being retired")
+    end
+
+    Notification.create!(:type => :vm_retiring, :subject => source)
```

Comment:
NOTE moved these common states up to the base MiqRetireTask class so they will apply to Orchestration stacks as well.

It is odd, but automate's Orchestration Retirement also uses `:vm_retiring`  notification type:
https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Cloud/Orchestration/Retirement/StateMachines/Methods.class/__methods__/start_retirement.rb#L48
But `:orchestration_stack_retired`
https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Cloud/Orchestration/Retirement/StateMachines/Methods.class/__methods__/finish_retirement.rb#L19

That is clearly a mistake, will fix separately